### DISCUSSION TITLE
What does the -u flag mean in git push -u origin master?

### DISCUSSION THREAD

**Discussion by rajshresth0** [Andipstha/drawIO#6] [Q&A] [answered]
I was just wondering what does the -u flag mean in this command?

git push -u origin master

**Comment by Andipstha** [comment, marked as answer]
The -u option does the following: For every branch that is up to date or successfully pushed, add an upstream (tracking) reference, used by argument-less, git-pull and other commands.

So, after pushing your local branch with the -u option, this local branch will be automatically linked with the remote branch, and you can use git pull without any arguments.
